src/Lumper.UI/ViewModels/MainWindowViewModel.IO.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Reactive.Concurrency;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Lumper.Lib.BSP;
+using Lumper.UI.ViewModels.Bsp;
+using MessageBox.Avalonia;
+using MessageBox.Avalonia.Enums;
+using ReactiveUI;
+
+namespace Lumper.UI.ViewModels;
+
+public partial class MainWindowViewModel
+{
+    private void IOInit()
+    {
+        this.WhenAnyValue(x => x.BspModel)
+            .ObserveOn(RxApp.MainThreadScheduler)
+            .Where(m => m is not null)
+            .Subscribe(x => BspModel!.RaisePropertyChanged(nameof(BspModel.FilePath)));
+
+        RxApp.MainThreadScheduler.Schedule(OnLoad);
+    }
+
+    private static OpenFileDialog ConstructOpenBspDialog()
```

Comment:
I'm getting warnings that "_Class 'Avalonia.Controls.OpenFileDialog' is obsolete: Use Window.StorageProvider API or TopLevel.StorageProvider API_" here, same with `SaveFileDialog`. Especially as we're just getting started with Avalonia, it seems keys we use the most up-to-date features.